util/reflect_test.go

```diff
@@ -847,3 +958,399 @@ func TestStructValueHasNFields(t *testing.T) {
 		}
 	}
 }
+
+type InnerContainerType1 struct {
+	LeafName *int32 `path:"leaf-field"`
+}
+type OuterContainerType1 struct {
+	Inner *InnerContainerType1 `path:"inner|config/inner"`
+}
+type ListElemStruct1 struct {
+	Key1   *string              `path:"key1"`
+	Outer  *OuterContainerType1 `path:"outer"`
+	Outer2 *OuterContainerType1 `path:"outer2"`
+}
+type ContainerStruct1 struct {
+	StructKeyList map[string]*ListElemStruct1 `path:"config/simple-key-list"`
+}
+
+func (*InnerContainerType1) IsYANGGoStruct() {}
```

Comment:
I think conventionally these should follow the definition of the type - rather than be in a block like this?

Reply:
Done.